From f91d256464bd0be921fd13cc182f990d3fa2696f Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 12 Dec 2012 16:32:09 +0800
Subject: [PATCH 386/635] ENGR00236726
 testRecorderTimelapsedVideo/testSetCamera failed

Align the default FPS RANGE to the default FPS in initParameters.
The logic to pick the FPS setting is choose the changed FPS or FPS RANGE
setting as the user request, which will be compatible with all API
setPreviewFpsRange or setPreviewFrameRate

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/Ov5640.cpp   | 5 +++--
 mx6/libcamera/Ov5642.cpp   | 5 +++--
 mx6/libcamera/OvDevice.cpp | 5 +++--
 3 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/mx6/libcamera/Ov5640.cpp b/mx6/libcamera/Ov5640.cpp
index 4403e53..2568b1a 100644
--- a/mx6/libcamera/Ov5640.cpp
+++ b/mx6/libcamera/Ov5640.cpp
@@ -169,7 +169,8 @@ status_t Ov5640::initParameters(CameraParameters& params,
                 mSupportedFPS);
     mParams.set(CameraParameters::KEY_SUPPORTED_PREVIEW_FPS_RANGE,
                 "(12000,17000),(25000,33000)");
-    mParams.set(CameraParameters::KEY_PREVIEW_FPS_RANGE, "25000,33000");
+    // Align the default FPS RANGE to the DEFAULT_PREVIEW_FPS
+    mParams.set(CameraParameters::KEY_PREVIEW_FPS_RANGE, "12000,17000");
 
     mParams.setPreviewSize(DEFAULT_PREVIEW_W, DEFAULT_PREVIEW_H);
     mParams.setPictureSize(DEFAULT_PICTURE_W, DEFAULT_PICTURE_H);
@@ -227,7 +228,7 @@ status_t Ov5640::setParameters(CameraParameters& params)
         return BAD_VALUE;
     }
 
-    local_framerate = params.getPreviewFrameRate();
+    local_framerate = mParams.getPreviewFrameRate();
     FLOGI("get local frame rate:%d FPS", local_framerate);
     if ((local_framerate > 30) || (local_framerate < 0)) {
         FLOGE("The framerate is not corrected");
diff --git a/mx6/libcamera/Ov5642.cpp b/mx6/libcamera/Ov5642.cpp
index 73aead8..5033f53 100644
--- a/mx6/libcamera/Ov5642.cpp
+++ b/mx6/libcamera/Ov5642.cpp
@@ -169,7 +169,8 @@ status_t Ov5642::initParameters(CameraParameters& params,
                 mSupportedFPS);
     mParams.set(CameraParameters::KEY_SUPPORTED_PREVIEW_FPS_RANGE,
                 "(12000,17000),(25000,33000)");
-    mParams.set(CameraParameters::KEY_PREVIEW_FPS_RANGE, "25000,33000");
+    // Align the default FPS RANGE to the DEFAULT_PREVIEW_FPS
+    mParams.set(CameraParameters::KEY_PREVIEW_FPS_RANGE, "12000,17000");
 
     mParams.setPreviewSize(DEFAULT_PREVIEW_W, DEFAULT_PREVIEW_H);
     mParams.setPictureSize(DEFAULT_PICTURE_W, DEFAULT_PICTURE_H);
@@ -227,7 +228,7 @@ status_t Ov5642::setParameters(CameraParameters& params)
         return BAD_VALUE;
     }
 
-    local_framerate = params.getPreviewFrameRate();
+    local_framerate = mParams.getPreviewFrameRate();
     FLOGI("get local frame rate:%d FPS", local_framerate);
     if ((local_framerate > 30) || (local_framerate < 0)) {
         FLOGE("The framerate is not corrected");
diff --git a/mx6/libcamera/OvDevice.cpp b/mx6/libcamera/OvDevice.cpp
index d1242c7..656051d 100644
--- a/mx6/libcamera/OvDevice.cpp
+++ b/mx6/libcamera/OvDevice.cpp
@@ -265,7 +265,8 @@ status_t OvDevice::initParameters(CameraParameters& params,
                 mSupportedFPS);
     mParams.set(CameraParameters::KEY_SUPPORTED_PREVIEW_FPS_RANGE,
                 "(12000,17000),(25000,33000)");
-    mParams.set(CameraParameters::KEY_PREVIEW_FPS_RANGE, "25000,33000");
+    // Align the default FPS RANGE to the DEFAULT_PREVIEW_FPS
+    mParams.set(CameraParameters::KEY_PREVIEW_FPS_RANGE, "12000,17000");
 
     mParams.setPreviewSize(DEFAULT_PREVIEW_W, DEFAULT_PREVIEW_H);
     mParams.setPictureSize(DEFAULT_PICTURE_W, DEFAULT_PICTURE_H);
@@ -319,7 +320,7 @@ status_t OvDevice::setParameters(CameraParameters& params)
         return BAD_VALUE;
     }
 
-    local_framerate = params.getPreviewFrameRate();
+    local_framerate = mParams.getPreviewFrameRate();
     FLOGI("get local frame rate:%d FPS", local_framerate);
     if ((local_framerate > 30) || (local_framerate < 0)) {
         FLOGE("The framerate is not corrected");
-- 
1.8.0

